src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,32 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+
+    Animal.alive.push(this);
+  }
+
+  static alive = [];
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hidden = false;
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(prey) {
+    if (!prey.hidden && prey instanceof Herbivore) {
+      prey.health -= 50;
+    }
+
+    Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
```

Comment:
Using `Array.filter` to remove dead animals from the `Animal.alive` array is a good approach and adheres to the checklist recommendation to avoid nesting loops.